src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java

```diff
@@ -77,10 +73,90 @@ public record VectorSearchConfig(
-      public VectorSearchConfig(Integer dimension, String metric) {
+            String metric,
+        @Valid
+            @Nullable
+            @JsonInclude(JsonInclude.Include.NON_NULL)
+            @Schema(
+                description = "Optional vectorize configuration to provide embedding service",
+                type = SchemaType.OBJECT,
+                implementation = VectorizeConfig.class)
+            @JsonProperty("service")
+            VectorizeConfig vectorizeConfig) {
+
+      public VectorSearchConfig(Integer dimension, String metric, VectorizeConfig vectorizeConfig) {
         this.dimension = dimension;
         this.metric = metric == null ? "cosine" : metric;
+        this.vectorizeConfig = vectorizeConfig;
+      }
+
+      public record VectorizeConfig(
+          @NotNull
+              @Schema(
+                  description = "Registered Embedding service provider",
+                  type = SchemaType.STRING,
+                  implementation = String.class)
+              @JsonProperty("provider")
+              String provider,
+          @NotNull
+              @Schema(
+                  description = "Registered Embedding service model",
+                  type = SchemaType.STRING,
+                  implementation = String.class)
+              @JsonProperty("model_name")
+              String modelName,
+          @Valid
+              @NotNull
+              @Schema(
+                  description = "Authentication config for chosen embedding service",
+                  type = SchemaType.OBJECT,
+                  implementation = VectorizeServiceAuthentication.class)
+              @JsonProperty("authentication")
+              VectorizeServiceAuthentication vectorizeServiceAuthentication,
+          @Valid
+              @JsonInclude(JsonInclude.Include.NON_NULL)
+              @Nullable
+              @Schema(
+                  description =
+                      "Optional parameters that match the template provided for the provider",
+                  type = SchemaType.OBJECT,
+                  implementation = VectorizeServiceParameter.class)
+              @JsonProperty("parameters")
+              VectorizeServiceParameter vectorizeServiceParameter) {
+        public record VectorizeServiceAuthentication(
+            @NotNull
+                @Schema(
+                    description =
+                        "List of authentications that can be used when sending documents that need vectorization. One or more of \"NONE\", \"HEADER\", \"SHARED_SECRET\"",
+                    type = SchemaType.ARRAY,
+                    implementation = String.class)
+                @JsonProperty("type")
+                List<
+                        @Pattern(
+                            regexp = "(NONE|HEADER|SHARED_SECRET)",
+                            message =
+                                "authentication type can only be one or more of 'NONE', 'HEADER' or 'SHARED_SECRET'")
```

Comment:
Need to have validation for this, based on vectorize configuration.

Reply:
captured in https://datastax.jira.com/jira/software/c/projects/C2/boards/525?assignee=712020%3Ac496b02d-1ca9-4999-845b-3b7b5cf1bc17&selectedIssue=C2-3312